spec/velveteen/worker_spec.rb

```diff
@@ -64,6 +64,57 @@
       _, properties, _ = queue.pop
       expect(properties[:headers]).to eq("foo" => "bar")
     end
+
+    describe "payload validation" do
+      it "raises an error when payload does not match schema" do
+        message = Velveteen::Message.new(
+          data: {"foo" => "bar"},
+          headers: {}
+        )
+        invalid_payload = {
+          data: {"coffee" => "espresso"},
+          headers: {}
+        }
+        worker = TestValidatesPayloadWorker.new(
+          message: message,
+          payload: invalid_payload
+        )
+        queue = Velveteen::Config.channel.queue
```

Comment:
Hmm, might not need to make this queue since exception will prevent anything from being published. What happens if it's removed?

Reply:
Still works! Good catch 🔍